pubnub/event_engine/manage_effects.py

```diff
@@ -28,51 +30,56 @@ def __init__(self, pubnub_instance, event_engine_instance,
         self.event_engine = event_engine_instance
         self.pubnub = pubnub_instance
 
+        self.logger = logging.getLogger("pubnub")
+
     def run(self):
         pass
 
     def run_async(self):
         pass
 
     def stop(self):
-        logging.debug(f'stop called on {self.__class__.__name__}')
         if self.stop_event:
-            logging.debug(f'stop_event({id(self.stop_event)}).set() called on {self.__class__.__name__}')
+            self.logger.debug(f'stop_event({id(self.stop_event)}).set() called on {self.__class__.__name__}')
             self.stop_event.set()
 
     def get_new_stop_event(self):
         event = asyncio.Event()
-        logging.debug(f'creating new stop_event({id(event)}) for {self.__class__.__name__}')
+        self.logger.debug(f'creating new stop_event({id(event)}) for {self.__class__.__name__}')
         return event
 
 
 class ManageHandshakeEffect(ManagedEffect):
     def run(self):
         channels = self.effect.channels
         groups = self.effect.groups
+        timetoken = self.effect.timetoken or 0
         if hasattr(self.pubnub, 'event_loop'):
             self.stop_event = self.get_new_stop_event()
 
             loop: asyncio.AbstractEventLoop = self.pubnub.event_loop
             if loop.is_running():
-                loop.create_task(self.handshake_async(channels, groups, self.stop_event))
+                loop.create_task(self.handshake_async(channels=channels, groups=groups, timetoken=timetoken,
+                                                      stop_event=self.stop_event))
```

Comment:
Nitpicking: Why not create the object before if to not repeat yourself with the whole argument list ;o?

Reply:
It's not an object, it's coroutine :) And you're right, I might change it in the future